server.go

```diff
@@ -361,6 +411,26 @@ type server struct {
 	// of new blocks.
 	blockbeatDispatcher *chainio.BlockbeatDispatcher
 
+	// banScoreMtx is used for the server's ban tracking. This should only
+	// be locked after the server mutex has been locked.
```

Comment:
Why use an alternative mutex if we already that expect the server mutex is locked?

Reply:
I'll change the comment, it's meant to convey that _if_ the server mutex is going to be locked, this mutex cannot be locked before the server mutex